packages/colorpickers/src/elements/ColorSwatchDialog/index.tsx

```diff
@@ -96,11 +98,23 @@ export const ColorSwatchDialog = forwardRef<
     const [uncontrolledSelectedColIndex, setUncontrolledSelectedColIndex] = useState(
       defaultSelectedColIndex || 0
     );
-    const uncontrolledSelectedColor =
-      colors[uncontrolledSelectedRowIndex][uncontrolledSelectedColIndex];
-    const controlledColor = isControlled
-      ? colors[selectedRowIndex as number][selectedColIndex as number]
-      : undefined;
+
+    let uncontrolledSelectedColor;
+    let controlledSelectedColorColor;
+
+    if (uncontrolledSelectedRowIndex > -1 && uncontrolledSelectedColIndex > -1) {
+      uncontrolledSelectedColor =
+        colors[uncontrolledSelectedRowIndex][uncontrolledSelectedColIndex];
+    }
+
+    if (
+      selectedRowIndex !== undefined &&
+      selectedColIndex !== undefined &&
+      selectedRowIndex > -1 &&
+      selectedColIndex > -1
+    ) {
```

Comment:
Alternatively could be `if (controlledSelect && selectedRowIndex > -1 && selectedColIndex > -1)` but unsure why TypeScript isn't liking that.